backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendClusterResource.java

```diff
@@ -73,10 +72,19 @@ public Response refreshGlusterHealStatus(Action action) {
 
     @Override
     public Response upgrade(Action action) {
-        if(action.getUpgradeAction() == ClusterUpgradeAction.START) {
-            return doAction(ActionType.StartClusterUpgrade, new ClusterParametersBase(guid), action);
+        switch (action.getUpgradeAction()) {
+            case START:
+                return doAction(ActionType.StartClusterUpgrade, new ClusterParametersBase(guid), action);
+
+            case UPDATE:
+                int progress = action.getUpgradeProgress();
```

Comment:
Looking at [this import action](http://ovirt.github.io/ovirt-engine-api-model/master/#services/storage_domain_vm/methods/import) might help: it has `collapse_snapshots` parameter defined in the [model here](https://github.com/oVirt/ovirt-engine-api-model/blob/1f9cc740aed35f672b6b3bdb7fd1b5b707cdc21a/src/main/java/services/StorageDomainVmService.java#L149).  It seems that this results in a getter in generated code: `org.ovirt.engine.api.model.Action.isCollapseSnapshots()`

Reply:
Looking at that example and the solution was to setup a new `ClusterUpgradeParameters` class and fill it out with the data from the action.  Current push of the PR works.